### PR TITLE
[DPE-2438] JupyterLab Spark integration

### PR DESCRIPTION
**Pull request by deusebio**
Works the same way but with JupyterLab instead of Notebook, which also seems a bit better as UX

### CODE REVIEW

File: tests/integration/resources/jupyter.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: Pod
+metadata:
+  name: spark-jupyter
+spec:
+  containers:
+  - image: ghcr.io/canonical/test-charmed-spark:3.4.1
+    name: spark
+    ports:
+    - containerPort: 8888
+    command: [pebble, enter, --args, jupyter, --namespace, spark, --username, spark-test ]
```
Comment (welpaolo) [inline tests/integration/resources/jupyter.yaml]: Is the username correct ?

Comment (welpaolo) [reply tests/integration/resources/jupyter.yaml]: Where are you using this pod ?

**Review by welpaolo — APPROVED** [review]
Looks good! I just left a couple of minor comments

File: rockcraft.yaml [hunk 2]
```diff
@@ -38,6 +38,14 @@ services:
...
     environment:
       SPARK_PROPERTIES_FILE: /etc/spark8t/conf/spark-defaults.conf
+  jupyter:
+    command: "spark-client.pyspark [ --username spark --namespace spark ]"
+    summary: "This is the Spark-powered Jupyter service"
+    override: replace
+    startup: disabled
+    environment:
+      PYSPARK_DRIVER_PYTHON: jupyter
+      PYSPARK_DRIVER_PYTHON_OPTS: "lab --no-browser --port=8888 --allow-root --ip=0.0.0.0 --NotebookApp.token='' --notebook-dir=/var/lib/spark/notebook"
```
Comment (grobbie) [inline rockcraft.yaml]: Is`--allow-root` needed? won't it break loadsa folks' security policies? Looks like it might?

**Review by grobbie — COMMENTED** [review]
open question on the `--allow-root` flag

File: tests/integration/resources/jupyter.yaml (hunk 1 above)
Comment (deusebio) [reply tests/integration/resources/jupyter.yaml]: Actually, we are not using this. I used it for testing, but that's not too relevant

File: rockcraft.yaml (hunk 2 above)
Comment (deusebio) [reply rockcraft.yaml]: removed this flag

**Comment by deusebio** [comment]
I have addressed the comments and rebase the branch. It should be ready for re-review

**Review by welpaolo — APPROVED** [review]

**Review by grobbie — APPROVED** [review]
thanks :+1:

**Pull request merged**